app/src/main/java/org/oppia/app/topic/practice/TopicPracticeFragmentPresenter.kt

```diff
@@ -5,97 +5,138 @@ import android.view.View
+        when (viewModel) {
+          is TopicPracticeHeaderViewModel -> ViewType.VIEW_TYPE_HEADER
+          is TopicPracticeSkillSummaryViewModel -> ViewType.VIEW_TYPE_SKILL
+          is TopicPracticeFooterViewModel -> ViewType.VIEW_TYPE_FOOTER
+          else -> throw IllegalArgumentException("Encountered unexpected view model: $viewModel")
+        }
+      }
+      .registerViewDataBinder(
+        viewType = ViewType.VIEW_TYPE_HEADER,
+        inflateDataBinding = TopicPracticeHeaderViewBinding::inflate,
+        setViewModel = TopicPracticeHeaderViewBinding::setViewModel,
+        transformViewModel = { it as TopicPracticeHeaderViewModel }
+      )
+      .registerViewDataBinder(
+        viewType = ViewType.VIEW_TYPE_SKILL,
+        inflateDataBinding = TopicPracticeSkillViewBinding::inflate,
+        setViewModel = this::bindSkillView,
+        transformViewModel = { it as TopicPracticeSkillSummaryViewModel }
+      )
+      .registerViewDataBinder(
+        viewType = ViewType.VIEW_TYPE_FOOTER,
+        inflateDataBinding = TopicPracticeFooterViewBinding::inflate,
+        setViewModel = this::bindFooterView,
+        transformViewModel = { it as TopicPracticeFooterViewModel }
+      )
+      .build()
   }
 
-  private fun getTopicList(): LiveData<Topic> {
-    return Transformations.map(topicResultLiveData, ::processTopicResult)
+  private fun bindSkillView(
+    binding: TopicPracticeSkillViewBinding,
+    model: TopicPracticeSkillSummaryViewModel
+  ) {
+    binding.viewModel = model
+    binding.isChecked = selectedSkillIdList.contains(model.skillSummary.skillId)
+    binding.skillCheckBox.setOnCheckedChangeListener { _, isChecked ->
+      if (isChecked) {
+        skillSelected(model.skillSummary.skillId)
+      } else {
+        skillUnselected(model.skillSummary.skillId)
+      }
+    }
   }
 
-  private fun processTopicResult(topic: AsyncResult<Topic>): Topic {
-    if (topic.isFailure()) {
-      logger.e("TopicPracticeFragment", "Failed to retrieve topic", topic.getErrorOrNull()!!)
-    }
-    return topic.getOrDefault(Topic.getDefaultInstance())
+  private fun bindFooterView(
+    binding: TopicPracticeFooterViewBinding,
+    model: TopicPracticeFooterViewModel
+  ) {
+    topicPracticeFooterViewBinding = binding
+    binding.viewModel = model
+    binding.isSubmitButtonActive = selectedSkillIdList.isNotEmpty()
   }
 
   private fun getTopicPracticeViewModel(): TopicPracticeViewModel {
     return viewModelProvider.getForFragment(fragment, TopicPracticeViewModel::class.java)
   }
 
+  private enum class ViewType {
+    VIEW_TYPE_HEADER,
+    VIEW_TYPE_SKILL,
+    VIEW_TYPE_FOOTER
+  }
+
   override fun skillSelected(skillId: String) {
     if (!selectedSkillIdList.contains(skillId)) {
       selectedSkillIdList.add(skillId)
     }
-    getTopicPracticeViewModel().notifySelectedSkillList(selectedSkillIdList)
+    topicPracticeFooterViewBinding.isSubmitButtonActive = selectedSkillIdList.isNotEmpty()
```

Comment:
Update:
```
    if (::topicPracticeFooterViewBinding.isInitialized) {
      topicPracticeFooterViewBinding.isSubmitButtonActive = selectedSkillIdList.isNotEmpty()
    }
```

We need to make sure that the footer has been initialised before working on its properties.

Reply:
Ok, I have applied the changes. code is working well. As an alternative of above change,       `topicPracticeFooterViewBinding?.isSubmitButtonActive = selectedSkillIdList.isNotEmpty()` this can also work right?